presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -72,12 +76,15 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
         double nullsCountLeft = leftStats.getNullsFraction() * leftRows;
         double totalSizeLeft = (leftRows - nullsCountLeft) * leftStats.getAverageRowSize();
         double totalSizeRight = (rightRows - nullsCountRight) * rightStats.getAverageRowSize();
-        double newNullsFraction = (nullsCountLeft + nullsCountRight) / newRowCount;
+        double newNullsFraction = newRowCount == 0 ? 1.0 : ((nullsCountLeft + nullsCountRight) / newRowCount);
         double newNonNullsRowCount = newRowCount * (1.0 - newNullsFraction);
 
+        // FIXME, weights to average. left and right should be equal in most cases anyway
```

Comment:
do you imply weighted average row size is wrong?

Reply:
I didn't add this comment. I just moved it.